tasks/verify-enterprise-contract.yaml

```diff
@@ -0,0 +1,123 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: verify-enterprise-contract
+  annotations:
+    tekton.dev/displayName: Verify Enterprise Contract
+    tekton.dev/pipelines.minVersion: "0.19"
+    tekton.dev/tags: cosign, chains, signature, opa
+  labels:
+    app.kubernetes.io/version: "0.1"
+
+spec:
+  description: Verify the enterprise contract is met
+  params:
+    - name: IMAGE_REF
+      type: string
+      description: Image reference to verify
+
+    - name: PUBLIC_KEY
+      type: string
+      description: >-
+        Public key used to verify signatures. Must be a valid k8s cosign
+        reference, e.g. k8s://my-space/my-secret where my-secret contains
+        the expected cosign.pub attribute.
+
+    - name: COSIGN_EXPERIMENTAL
+      type: string
+      description: Control transparency log lookups. Set to "1" to enabled it.
+      default: "1"
+
+    - name: REKOR_HOST
+      type: string
+      description: Rekor host for transparency log lookups
+      default: https://rekor.sigstore.dev
+
+    # TODO: We should probably use a bundle URL for this:
+    # https://www.openpolicyagent.org/docs/v0.12.2/bundles/#bundle-file-format
+    # Or an OCI image:
+    # https://github.com/open-policy-agent/opa/issues/1413
+    - name: POLICY_REPO
+      type: string
+      description: Git url for rego policies
+      default: https://github.com/hacbs-contract/ec-policies.git
+
+    - name: POLICY_REPO_REF
+      type: string
+      description: Git sha, branch or tag in git repo for rego policies
+      default: main
+
+  results:
+    - name: OUTPUT
+      description: A list of policy violations
+    - name: PASSED
+      description: A string formatted boolean, either true or false
+
+  steps:
+    - name: verify-image-signature
+      image: appstudio-utils
+      command: [cosign]
+      args:
+        - verify
+        - "--key=$(params.PUBLIC_KEY)"
+        - "--rekor-url=$(params.REKOR_HOST)"
+        - $(params.IMAGE_REF)
+      env:
+        # TODO: What if the workspace is not set? Seems to have no effect:
+        #   https://issues.redhat.com/browse/HACBS-316
+        # TODO: Is it SSL_CERT_DIR or SSL_CERTS_DIR ?
+        - name: SSL_CERT_DIR
```

Comment:
Should be [`SSL_CERT_DIR`](https://github.com/golang/go/blob/96c8cc7fea94dca8c9e23d9653157e960f2ff472/src/crypto/x509/root_unix.go#L25)

Reply:
Cool. I found `SSL_CERT_DIR` in the docs as well but was confused because the `cosign-verify*` tasks use `SSL_CERTS_DIR`.

Reply:
Good catch! That's a typo there, a [pr to fix it](https://github.com/redhat-appstudio/build-definitions/pull/70).